storage/aptosdb/src/state_store/mod.rs

```diff
@@ -706,6 +720,22 @@ impl StateStore {
                     .collect::<Result<_>>()
             })?;
 
+        // Eventually this index will move to indexer side. For now we temporarily write this into
+        // metadata db to unblock the sharded DB migration.
+        // TODO(grao): Remove when we are ready.
+        if put_state_value_indices {
+            value_state_sets
+                .par_iter()
+                .enumerate()
+                .try_for_each(|(i, updates)| {
+                    let version = first_version + i as Version;
+                    updates.iter().flatten().try_for_each(|(k, _)| {
+                        state_kv_metadata_batch
+                            .put::<StateValueIndexSchema>(&(k.clone(), version), &())
```

Comment:
what is the purpose of having empty value &()?

Reply:
It needs to have something as the value.

Reply:
I see. where is the value populated? I was not able to find the place in the PR.

Reply:
Here. we fill an empty thing "&()" as the value, because we don't need a value.

Reply:
the key only has stateKey and version. how do we get the state value? I assume the purpose is to store all state values in one schema to "read everything by key prefix".

Reply:
state value is still stored in the old "StateValueSchema" (sharded). We get all keys from the StateValueIndexSchema, then use those keys to lookup the value in StateValueSchema.